piptools/repositories/pypi.py

```diff
@@ -452,7 +461,7 @@ def _setup_logging(self) -> None:
         logger = logging.getLogger()
         for handler in logger.handlers:
             if handler.name == "console":  # pragma: no branch
-                assert isinstance(handler, logging.StreamHandler)
+                assert isinstance(handler, compatible_handler)
```

Comment:
I think it's going to take more than this. The RichPipStreamHandler sends its output to a rich `console`, not to a `stream`.

It looks like this code was added in https://github.com/jazzband/pip-tools/pull/949 to fix a problem with displaying the progress bars from pip's verbose output.

But it appears that that fix is no longer needed when the progress bars are rendered using `rich`. I tested and if I simply remove lines 451-466 and run pip-compile in verbose mode, the progress bars display fine and the command completes successfully with pip 22.

So I would suggest making these lines conditional on whether RichPipStreamHandler was imported above.

Reply:
Expected something like this would come up. Thanks for checking! Sounds like a reasonable solution, while I'm not sure if the whole code introduced in #949 can be "reverted / ignored" depending on pip version or we should only skip the assignment of the "proper" stream to the respective handler during execution.

@atugushev would be happy to have your expertise here as you worked on adding this back then in case you have time.